ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/DownloadHeadersStep.java

```diff
@@ -85,23 +95,52 @@ private CompletableFuture<List<BlockHeader>> downloadHeaders(final SyncTargetRan
               protocolSchedule,
               protocolContext,
               ethContext,
+              peerTaskExecutor,
+              synchronizerConfiguration,
               range.getEnd(),
               range.getSegmentLengthExclusive(),
               validationPolicy,
               metricsSystem)
           .run();
     } else {
       LOG.debug("Downloading headers starting from {}", range.getStart().getNumber());
-      return GetHeadersFromPeerByHashTask.startingAtHash(
-              protocolSchedule,
-              ethContext,
-              range.getStart().getHash(),
-              range.getStart().getNumber(),
-              headerRequestSize,
-              metricsSystem)
-          .assignPeer(range.getSyncTarget())
-          .run()
-          .thenApply(PeerTaskResult::getResult);
+      if (synchronizerConfiguration.isPeerTaskSystemEnabled()) {
+        return ethContext
+            .getScheduler()
+            .scheduleServiceTask(
+                () -> {
+                  LOG.info("Getting headers using peer task system");
```

Comment:
change at least to debug!

Reply:
I'll remove these logs. I was just using them to check that the code was working